internal/agent/device/spec/spec.go

```diff
@@ -266,7 +242,29 @@ func EnsureDesiredRenderedSpec(
 	return desired, nil
 }
 
-func writeRenderedSpecToFile(writer *fileio.Writer, rendered *v1alpha1.RenderedDeviceSpec, filePath string) error {
+func ReadRenderedSpecFromFile(
+	reader *fileio.Reader,
+	filePath string,
+) (v1alpha1.RenderedDeviceSpec, error) {
+	var current v1alpha1.RenderedDeviceSpec
+	renderedBytes, err := reader.ReadFile(filePath)
```

Comment:
lets add a TODO to use buffered read. these are perf optimizations but we should be tracking them as these files could get fairly large potentially. I know your just moving it but its on my list.